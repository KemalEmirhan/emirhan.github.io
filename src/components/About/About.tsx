import React from 'react';
import useAge from '@/hooks/useAge';

const About = (): React.ReactElement => {
  const age = useAge();

  return (
    <article className='max-w-4xl mx-auto px-6 py-12'>
      <h2 className='text-3xl font-bold text-gray-900 mb-8'>About Me</h2>
      <section className='mb-8'>
        <p className='text-lg text-gray-700 leading-relaxed mb-4'>
          Hey, I'm Emirhan ðŸ‘‹ &nbsp; {age} years old. I'm Front-End Software
          Engineer who writes HTML5, CSS3 and JS(reactjs, nextjs). I'm
          interested with web standards, a11y and Figma these days. <wbr />
        </p>
        <p className='text-lg text-gray-700 leading-relaxed'>
          If you reach me, you feel free âœŒ
        </p>
      </section>
      <section>
        <p className='text-lg text-gray-700 leading-relaxed'>
          Front-End Software Engineer @Iyzico
        </p>
      </section>
    </article>
  );
};

export default About;
