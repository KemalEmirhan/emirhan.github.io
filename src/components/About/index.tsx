'use client';

import React, { useMemo } from 'react';
import { motion, Easing } from 'motion/react';
import useAge from '@/hooks/useAge';

const About = (): React.ReactElement => {
  const age = useAge();

  const animationProps = useMemo(
    () => ({
      initial: { opacity: 0, y: 20 },
      animate: { opacity: 1, y: 0 },
      exit: { opacity: 0, y: 20 },
      transition: { duration: 0.2, ease: 'easeOut' as Easing },
    }),
    []
  );

  return (
    <motion.article
      className='space-y-6 text-base md:text-lg text-gray-700 leading-relaxed max-w-none'
      {...animationProps}
    >
      <h2 className='sr-only'>About Me</h2>
      <motion.p>
        Hey, I'm Emirhan ðŸ‘‹ {age} years old. I'm a Frontend Software Engineer
        with over 8 years of experience building high-performance web
        applications across various industries. Based in Berlin, I specialise in
        React, Next.js, JavaScript, TypeScript, and GraphQL, with a strong focus
        on accessibility and web standards.
      </motion.p>
      <motion.p>
        Throughout my career in fintech, media, telecommunications, and
        e-commerce, I've collaborated closely with cross-functional teams to
        develop scalable solutions. I enjoy modernising outdated systems,
        designing efficient component libraries that accelerate development, and
        optimising web applications for seamless user experiences. By combining
        technical expertise with a strategic mindset, I make architectural
        decisions that enhance performance and drive business success.
      </motion.p>
      <motion.p>
        Passionate about creating accessible, user-friendly digital experiences,
        I'm always eager to stay ahead of the latest software technologies and
        continuously expand my skill set.
      </motion.p>
    </motion.article>
  );
};

export default About;
