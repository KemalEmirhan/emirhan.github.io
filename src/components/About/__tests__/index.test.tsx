import { describe, expect, test, vi } from 'vitest';
import { render } from '@testing-library/react';
import About from '../';

vi.mock('@/hooks/useAge', () => ({
  default: () => 30,
}));

describe('About Component', () => {
  test('should render', () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();
  });
});
