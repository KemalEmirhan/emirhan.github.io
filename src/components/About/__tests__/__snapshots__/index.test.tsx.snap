// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About Component > should render 1`] = `
<div>
  <article
    class="flex flex-col gap-4 text-base md:text-lg text-gray-900 leading-relaxed max-w-none"
    style="opacity: 0; transform: translateY(20px);"
  >
    <h2
      class="sr-only"
    >
      About Me
    </h2>
    <p>
      Hey, I'm Emirhan ðŸ‘‹ 
      30
       years old. I'm a Frontend Software Engineer with over 8 years of experience building high-performance web applications across various industries. Based in Berlin, I specialise in React, Next.js, JavaScript, TypeScript, and GraphQL, with a strong focus on accessibility and web standards.
    </p>
    <p>
      Throughout my career in fintech, media, telecommunications, and e-commerce, I've collaborated closely with cross-functional teams to develop scalable solutions. I enjoy modernising outdated systems, designing efficient component libraries that accelerate development, and optimising web applications for seamless user experiences. By combining technical expertise with a strategic mindset, I make architectural decisions that enhance performance and drive business success.
    </p>
    <p>
      Passionate about creating accessible, user-friendly digital experiences, I'm always eager to stay ahead of the latest software technologies and continuously expand my skill set.
    </p>
  </article>
</div>
`;
